xbmc/platform/darwin/ios-common/AnnounceReceiver.mm

```diff
@@ -168,6 +170,23 @@ void AnnounceBridge(ANNOUNCEMENT::AnnouncementFlag flag,
         item[@"genre"] = genreArray;
       }
     }
+    if (g_application.CurrentFileItem().IsPVRChannel())
+    {
+      auto epg_now = g_application.CurrentFileItem().GetPVRChannelInfoTag()->GetEPGNow();
+      item[@"epg_title"] = @(epg_now->Title().c_str());
+      auto epg_icon = epg_now->Icon();
+      if (!epg_icon.empty())
+      {
+        bool needsRecaching;
+        std::string cachedIcon(CTextureCache::GetInstance().CheckCachedImage(epg_icon, needsRecaching));
+        if (cachedIcon.empty())
+        {
+          cachedIcon = CTextureCache::GetInstance().CacheImage(epg_icon);
+        }
+        std::string iconRealPath = CSpecialProtocol::TranslatePath(cachedIcon);
+        item[@"epg_icon"] = @(iconRealPath.c_str());
+      }
+    }
```

Comment:
This block reads as a duplicate of the thumb setup. Would it not be easier/better to just add a check for pvrchannel in an if/else around the following line so you dont duplicate the whole cache check process? Then no need for the new dict item epg_icon, as thumb will be either the existing thumb, or the epg icon?
https://github.com/xbmc/xbmc/blob/9242ebc46341982f145d05897b35f39ec28de244/xbmc/platform/darwin/ios-common/AnnounceReceiver.mm#L137

Reply:
Something like this maybe

```
--- a/xbmc/platform/darwin/ios-common/AnnounceReceiver.mm
+++ b/xbmc/platform/darwin/ios-common/AnnounceReceiver.mm
@@ -132,7 +132,18 @@ void AnnounceBridge(ANNOUNCEMENT::AnnouncementFlag flag,
     NSMutableDictionary* item = dict[@"item"];
     NSDictionary* player = dict[@"player"];
     item[@"speed"] = player[@"speed"];
-    std::string thumb = g_application.CurrentFileItem().GetArt("thumb");
+    std::string thumb;
+    if (g_application.CurrentFileItem().IsPVRChannel())
+    {
+      auto epg_now = g_application.CurrentFileItem().GetPVRChannelInfoTag()->GetEPGNow();
+      item[@"epg_title"] = @(epg_now->Title().c_str());
+      thumb = epg_now->Icon();
+    }
+    else
+    {
+      thumb = g_application.CurrentFileItem().GetArt("thumb");
+    }
+
     if (!thumb.empty())
     {
       bool needsRecaching;
```

Reply:
Work it as needed. IE, maybe you want a fallback to the getart("thumb"), so in that case, do the thumb assignment as is in current code, do the if pvrchannel, check for length of epg_now->Icon() > 0, then assign thumb to it or something to that effect. then if there is no epg icon, you still have the getart thumb icon fallback